database/src/pool/postgres.rs

```diff
@@ -254,6 +254,15 @@ static MIGRATIONS: &[&str] = &[
         UNIQUE(aid, component)
     );
     "#,
+    // Add codegen backend column and add it to the unique constraint.
+    // Also rename cache to scenario and statistic to metric, while we're at it.
+    r#"
+    alter table pstat_series rename column cache to scenario;
+    alter table pstat_series rename column statistic to metric;
+    alter table pstat_series add backend text not null default 'llvm';
+    alter table pstat_series drop constraint pstat_series_crate_profile_cache_statistic_key;
+    alter table pstat_series add constraint test_case UNIQUE(crate, profile, scenario, backend, metric);
```

Comment:
Have we tested this migration on a local database or so? I'm a bit worried that there's (AFAICT) no index that would help with this unique constraint, right? Or if there is such an index, then maybe we need to drop the *previous* index so we're not ending up with 2 separate indices.

Creating the new index may also be pretty slow, but I suppose that's a one-time cost and shouldn't be too bad.

Reply:
Hmm, I kind of assumed that dropping the UNIQUE constraint would also drop the corresponding index. When I tried it on a local Postgres database, the old index was removed. I tried it again now. I converted the live site data from SQLite to Postgres and then ran the migration.

The migration took 63 milliseconds, for ~13 thousand `pstat_series` entries. I guess that there's not *that much* entries in this table, so it shouldn't take too long, hopefully.

Before:
![image](https://github.com/rust-lang/rustc-perf/assets/4539057/204d9c7b-cf24-4987-a9e0-450371ed6902)

After:
![image](https://github.com/rust-lang/rustc-perf/assets/4539057/6d009c5a-9ba0-4fbd-afe3-7825e287025a)

Reply:
Hm, ok. I guess it's auto-created? Should be OK then.